From 7ee483f2f923d1aebcb5913d1c545be2df289b61 Mon Sep 17 00:00:00 2001
From: ArenM <aren@peacevolution.org>
Date: Fri, 5 May 2023 16:33:11 -0400
Subject: [PATCH sxmo-utils 3/6] Manual fixups for nerd fonts 3.0

* Make icon_prn a nerd-font icons. It was using a font awesome code
  point, which we don't install with sxmo.
* Fix phone locked and laptop icons that weren't automatically updated
* Fix brightness icon in legacy nerd-fonts range and move it to
  sxmo_hook_icons from sxmo_brightness
* Use standard Unicode arrows instead of nerd-fonts ones
* Use icon variables for arrows in sxmo_wmmenu.sh
* Add variable for audio icon & use in sxmo_audio.sh
* Remove no stray break space from sxmo_contactmenu
* Remove up arrow character from reddit script
* use only icon variables in modemtext
---
 configs/default_hooks/sxmo_hook_icons.sh | 24 +++++++++++++-----------
 scripts/appscripts/sxmo_reddit.sh        |  2 +-
 scripts/core/sxmo_audio.sh               |  2 +-
 scripts/core/sxmo_brightness.sh          |  6 ++++--
 scripts/core/sxmo_contactmenu.sh         |  2 +-
 scripts/core/sxmo_wmmenu.sh              |  6 +++---
 scripts/modem/sxmo_modemtext.sh          | 14 ++++++++++----
 7 files changed, 33 insertions(+), 23 deletions(-)

diff --git a/configs/default_hooks/sxmo_hook_icons.sh b/configs/default_hooks/sxmo_hook_icons.sh
index c86fac8..8692f97 100755
--- a/configs/default_hooks/sxmo_hook_icons.sh
+++ b/configs/default_hooks/sxmo_hook_icons.sh
@@ -14,18 +14,20 @@
 
 icon_a2x="⇄"
 icon_a2y="⇅"
-icon_ac1="󰁛"
-icon_ac2="󰁜"
-icon_ac3="󰁃"
-icon_ac4="󰁂"
+icon_ac1="↖"
+icon_ac2="↗"
+icon_ac3="↘"
+icon_ac4="↙"
 icon_and=""
-icon_ard="󰁅"
-icon_arl="󰁍"
-icon_arr="󰁔"
-icon_aru="󰁝"
+icon_ard="↓"
+icon_arl="←"
+icon_arr="→"
+icon_aru="↑"
 icon_att=""
+icon_audio="♫"
 icon_bel=""
 icon_bok=""
+icon_brightness="󰃝"
 icon_bth="" # bluetooth
 icon_cal=""
 icon_cam=""
@@ -35,7 +37,7 @@ icon_chs="♜" #chess
 icon_clc="󰃬" #calculator
 icon_clk="󰥔"
 icon_cls="󰅖" #close
-icon_com="" # (Laptop) Computer (💻)
+icon_com="" # (Laptop) Computer (💻)
 icon_cpy=""
 icon_del="󰅖" #delete
 icon_dir="" #directory folder
@@ -91,9 +93,9 @@ icon_pau=""
 icon_phl="󰏹" #phonelog
 icon_phn="󰏲" #phone
 icon_phx="󰏵" #hangup
-icon_plk="" # phone locked
+icon_plk="󰏸" # phone locked
 icon_pls=""
-icon_prn="🖶" # printer (🖨️)
+icon_prn="" # printer (🖨️)
 icon_prv="" #previous
 icon_pst=""
 icon_pwr="⏻" #power
diff --git a/scripts/appscripts/sxmo_reddit.sh b/scripts/appscripts/sxmo_reddit.sh
index 0c778c5..55ef2b3 100755
--- a/scripts/appscripts/sxmo_reddit.sh
+++ b/scripts/appscripts/sxmo_reddit.sh
@@ -23,7 +23,7 @@ menu() {
 			sed 's/===/\n/g' |
 			sed 's/^\t//g' |
 			sort -t"$(printf '%b' '\t')" -rnk4 |
-			awk -F'\t' '{ printf "↑%4s", $3; print " " $4 " " $1 " " $2 }'
+			awk -F'\t' '{ printf "%4s", $3; print " " $4 " " $1 " " $2 }'
 	)"
 
 	while true; do
diff --git a/scripts/core/sxmo_audio.sh b/scripts/core/sxmo_audio.sh
index 23eb06a..0b7fcb0 100755
--- a/scripts/core/sxmo_audio.sh
+++ b/scripts/core/sxmo_audio.sh
@@ -19,7 +19,7 @@ notifyvol() {
 	if [ "$SXMO_WM" = "sway" ] && [ -z "$SXMO_WOB_DISABLE" ]; then
 		printf "%s\n" "$vol" > "$XDG_RUNTIME_DIR"/sxmo.wobsock
 	else
-		dunstify -r 999 "♫ Volume $vol"
+		dunstify -r 999 "$icon_audio Volume $vol"
 	fi
 }
 
diff --git a/scripts/core/sxmo_brightness.sh b/scripts/core/sxmo_brightness.sh
index 4c25fb5..aec0002 100755
--- a/scripts/core/sxmo_brightness.sh
+++ b/scripts/core/sxmo_brightness.sh
@@ -4,13 +4,15 @@
 
 # include common definitions
 # shellcheck source=scripts/core/sxmo_common.sh
-. "$(dirname "$0")/sxmo_common.sh"
+. sxmo_common.sh
+# shellcheck source=configs/default_hooks/sxmo_hook_icons.sh
+. sxmo_hook_icons.sh
 
 notify() {
 	if [ "$SXMO_WM" = "sway" ] && [ -z "$SXMO_WOB_DISABLE" ]; then
 		light | grep -o "^[0-9]*" > "$XDG_RUNTIME_DIR"/sxmo.wobsock
 	else
-		light | xargs dunstify -r 888 " Brightness"
+		light | xargs dunstify -r 888 "$icon_brightness Brightness"
 	fi
 }
 
diff --git a/scripts/core/sxmo_contactmenu.sh b/scripts/core/sxmo_contactmenu.sh
index aade036..225a42d 100755
--- a/scripts/core/sxmo_contactmenu.sh
+++ b/scripts/core/sxmo_contactmenu.sh
@@ -80,7 +80,7 @@ deletecontact() {
 	ENTRIES="$(printf "$icon_cls No\n$icon_chk Yes")"
 	PICKED="$(
 		printf %b "$ENTRIES" |
-		dmenu -p "$icon_del Delete $name ?"
+		dmenu -p "$icon_del Delete $name?"
 	)"
 
 	# reverse them
diff --git a/scripts/core/sxmo_wmmenu.sh b/scripts/core/sxmo_wmmenu.sh
index fbe4228..3d40b31 100755
--- a/scripts/core/sxmo_wmmenu.sh
+++ b/scripts/core/sxmo_wmmenu.sh
@@ -76,11 +76,11 @@ swaywmmenu() {
 		)"
 		if [ -n "$CURRENT_LAYOUT" ]; then
 			if [ "$CURRENT_LAYOUT" = "splith" ]; then
-				LAYOUT_LINE="splith → splitv → tabbed"
+				LAYOUT_LINE="splith $icon_arr splitv $icon_arr tabbed"
 			elif [ "$CURRENT_LAYOUT" = "tabbed" ] ; then
-				LAYOUT_LINE="tabbed → splith → splitv"
+				LAYOUT_LINE="tabbed $icon_arr splith $icon_arr splitv"
 			else
-				LAYOUT_LINE="splitv → tabbed → splith"
+				LAYOUT_LINE="splitv $icon_arr tabbed $icon_arr splith"
 			fi
 		fi
 		CHOICES="$(grep . <<EOF
diff --git a/scripts/modem/sxmo_modemtext.sh b/scripts/modem/sxmo_modemtext.sh
index e4ca430..976519b 100755
--- a/scripts/modem/sxmo_modemtext.sh
+++ b/scripts/modem/sxmo_modemtext.sh
@@ -65,19 +65,25 @@ sendtextmenu() {
 
 	while true
 	do
+		# We use them in printf statements
+		export icon_cls
+		export icon_att
+		export icon_usr
+
 		ATTACHMENTS=
 		if [ -f "$SXMO_LOGDIR/$NUMBER/draft.attachments.txt" ]; then
+			# TODO: test this
 			# shellcheck disable=SC2016
 			ATTACHMENTS="$(
 				tr '\n' '\0' < "$SXMO_LOGDIR/$NUMBER/draft.attachments.txt" |
-				xargs -0 -I{} sh -c 'printf "%s %s %s :: %s\n" "" "" "$(basename "{}")" "{}"'
+				xargs -0 -I{} sh -c 'printf "%s %s %s :: %s\n" "$icon_cls" "$icon_att" "$(basename "{}")" "{}"'
 			)"
 		fi
 
 		RECIPIENTS=
 		if [ "$(printf %s "$NUMBER" | xargs pnc find | wc -l)" -gt 1 ]; then
 			# shellcheck disable=SC2016
-			RECIPIENTS="$(printf %s "$NUMBER" | xargs pnc find | xargs -I{} sh -c 'printf "  "$(sxmo_contacts.sh --name {})" :: {}\n"')"
+			RECIPIENTS="$(printf %s "$NUMBER" | xargs pnc find | xargs -I{} sh -c 'printf "$icon_cls $icon_usr "$(sxmo_contacts.sh --name {})" :: {}\n"')"
 		fi
 
 		CHOICES="$(printf "%s Send to %s (%s)\n%b\n%s Add Recipient\n%b\n%s Add Attachment\n%s Edit '%s'\n%s Cancel\n" \
@@ -98,7 +104,7 @@ sendtextmenu() {
 				fi
 				;;
 			# Remove Attachment
-			" "*)
+			"$icon_cls $icon_att"*)
 				FILE="$(printf %s "$CONFIRM" | awk -F' :: ' '{print $2}')"  
 				sed -i "\|$FILE|d" "$SXMO_LOGDIR/$NUMBER/draft.attachments.txt"
 				if [ ! -s "$SXMO_LOGDIR/$NUMBER/draft.attachments.txt" ] ; then
@@ -106,7 +112,7 @@ sendtextmenu() {
 				fi
 				;;
 			# Remove Recipient
-			" "*)
+			"$icon_cls $icon_usr"*)
 				if [ "$(printf %s "$NUMBER" | xargs pnc find | wc -l)" -gt 1 ]; then
 					OLDNUMBER="$NUMBER"
 					RECIPIENT="$(printf %s "$CONFIRM" | awk -F' :: ' '{print $2}')"
-- 
2.40.0

